src/entry-point-client/index.tsx

```diff
@@ -1,20 +1,51 @@
+import '@babel/polyfill';
+
 import React from "react";
 (window as any).React = React;
 
 import ReactDOM from "react-dom";
-import { BrowserRouter } from "react-router-dom";
+import { Router } from "react-router-dom";
 
 import App from "./App";
 
+import * as serviceWorker from "../serviceWorker";
+import { Auth0Provider } from "../react-auth0-spa";
+import config from "../auth_config.json";
+import history from "../utils/history";
+
+// A function that routes the user to the right place
+// after login
+const onRedirectCallback = appState => {
+  history.push(
+    appState && appState.targetUrl
+      ? appState.targetUrl
+      : window.location.pathname
+  );
+};
+
 let element = (
-  <BrowserRouter>
-    <App />
-  </BrowserRouter>
+  <Auth0Provider
+    domain={config.domain}
+    client_id={config.clientId}
+    audience={config.audience}
+    redirect_uri={window.location.origin}
+    onRedirectCallback={onRedirectCallback}
+  >
+    <Router history={history}>
```

Comment:
We needed to switch from `BrowserRouter` to `Router` with the `history` attribute to support proper auth'd navigation back and forth.